crates/p2p/src/swarm/mod.rs

```diff
@@ -432,20 +429,32 @@ where
             }
             Command::RequestMessage(request) => {
                 let request_target_pubkey = request.operator_pubkey();
-                let distributor_peer_id = self
+
+                let maybe_distributor = self
                     .db
                     .get_peer_by_signer_pubkey(request_target_pubkey)
                     .await?;
 
-                let Some(distributor_peer_id) = distributor_peer_id else {
-                    warn!("Tried to sent request for operator that has no corresponding peer");
-                    return Ok(());
-                };
+                let request = request.into_msg();
+
+                if let Some(distributor_peer_id) = maybe_distributor {
+                    if self.swarm.is_connected(&distributor_peer_id) {
+                        self.swarm
+                            .behaviour_mut()
+                            .request_response
+                            .send_request(&distributor_peer_id, request);
+                        return Ok(());
+                    } // TODO: try to establish connection?
+                }
+
+                let connected_peers = self.swarm.connected_peers().cloned().collect::<Vec<_>>();
+                for peer in connected_peers {
```

Comment:
You don't need to allocate `Vec` here

```suggestion
                for peer in  self.swarm.connected_peers() {
```